src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -1663,7 +1663,12 @@ NameToNameVector MergeTreeDataMergerMutator::collectFilesForRenames(
     {
         if (command.type == MutationCommand::Type::DROP_INDEX)
         {
-            if (source_part->checksums.has(INDEX_FILE_PREFIX + command.column_name + ".idx"))
+            if (source_part->checksums.has(INDEX_FILE_PREFIX + command.column_name + ".idx2"))
```

Comment:
I wonder when we are going to have proper metadata instead of this. With columns, the situation is already becoming unmanageable, I think Anton is going to add something like "serialization.txt" in his PR for sparse columns, which has the necessary metadata. Not sure it applies to the indexes though.

Reply:
I was thinking about header. But I decided not to put it here since this will complicate the patch.
Note that serialization.txt will not solve such hunks automatically (you still need to know all the extension here, or add some interface into serializer to return them), while header will.